### PR TITLE
feat(node): add child_process.fork

### LINKED ISSUE #1690: node: add child_process.fork
child_process.fork is used by ava. ref: https://github.com/avajs/ava/blob/77623a5f32c139fa4ef7d316a4bbf7e172cc5acb/lib/fork.js#L51

### PR DESCRIPTION
**Pull request by kt3k**
This PR implements child_process.fork in `std/node`

This implements at least the basic feature of `fork` and enabled the basic test cases (`child-process-spawn-node.js`).

This doesn't implement `ipc` features. So I commented out ipe-related assertion in the test cases.

The diff might look big, but that is because this PR moves `ChildProcess` class from `node/child_process` to `node/internal/chil_process` to follow the internal structure of Node.js code base.

closes #1690

### CODE REVIEW

**Comment by kt3k** [comment]
The CI is blocked by #1694

**Review by bartlomieju — COMMENTED** [review]
@kt3k is this ready? I don't see any new method in the `chil_process.ts` file.

**Comment by kt3k** [comment]
@bartlomieju This should add `fork` method. Please see the last line of `node/child_process.ts`. BTW now we have two `child_process.ts` files in `node/` and `node/internal/`.

File: node/child_process.ts [hunk 1]
```diff
@@ -2,299 +2,98 @@
...
+  ERR_INVALID_ARG_VALUE,
+} from "./_errors.ts";
+import { process } from "./process.ts";
+
+const denoCompatArgv = [
+  "run",
+  "--compat",
+  "--unstable",
+  "--no-check",
+  "--location=https://example.com",
```
Comment (bartlomieju) [inline node/child_process.ts:22]: Why is `location` argument needed?

File: node/child_process.ts [hunk 2]
```diff
@@ -2,299 +2,98 @@
...
+  "--allow-all",
```
Comment (bartlomieju) [inline node/child_process.ts:23]: Yeah... this is problematic, ideally we'd use the same permissions as the parent process, but unfortunately we don't have access to these flags.

**Review by bartlomieju — COMMENTED** [review]
Thanks @kt3k I was looking at the wrong file. This API is gonna be tricky due to flags, but this is good progress

File: node/child_process.ts (hunk 1 above)
Comment (kt3k) [reply node/child_process.ts:22]: Some npm module does something like `typeof location` to detect the environment and causes uncaught errors. So giving `--location` avoids those errors, but that probably should be solved by https://github.com/denoland/deno/issues/12969

Ok. I'll remove this option as a first pass

Comment (bartlomieju) [reply node/child_process.ts]: Yeah, let's address it in Deno directly

**Comment by kt3k** [comment]
Oh, because we use `deno run --compat` for spawning subcommand, the subprocess have to use `std@0.117.0` for std/node and which still includes `BINARY` notation in enum..

**Comment by bartlomieju** [comment]
Unfortunately I don't see a good way to fix it at the moment, as `deno.land/std/` doesn't allow to link to `main` branch

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -100,6 +100,8 @@ jobs:
 
       - name: Run Node tests canary
         run: deno test --unstable --allow-all --import-map=test_import_map.json --config strict-ts44.tsconfig.json node/_tools/test.ts
+        env:
+          DENO_NODE_COMPAT_URL: file:///${{ github.workspace }}/
```
Comment (bartlomieju) [inline .github/workflows/ci.yml:104]: This is a good workaround for now

**Review by bartlomieju — APPROVED** [review]
LGTM

**Pull request merged**